  INFO root [2020-10-24 23:00:38,274] - 
  INFO root [2020-10-24 23:00:38,274] - -----------------------------------------------------------------------------------------


  INFO root [2020-10-24 23:00:38,275] - Resmgr::initialize: KBE_ROOT=D:/kbengine_24pic/
  INFO root [2020-10-24 23:00:38,276] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine_24pic/kbe/res/;D:/kbengine_24pic/kbengine_stone_assets/;D:/kbengine_24pic/kbengine_stone_assets/scripts/;D:/kbengine_24pic/kbengine_stone_assets/res/
  INFO root [2020-10-24 23:00:38,276] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine_24pic/kbe/bin/server/
  INFO root [2020-10-24 23:00:38,283] - NetworkInterface::initialize(EXTERNAL): bound to all interfaces with default route interface on 192.168.2.246 ( 192.168.2.246:20013 )
  INFO root [2020-10-24 23:00:38,284] - NetworkInterface::initialize(EXTERNAL): address 192.168.2.246:20013, SOMAXCONN=128.
  INFO root [2020-10-24 23:00:38,287] - NetworkInterface::initialize(INTERNAL): bound to all interfaces with default route interface on 192.168.2.246 ( 192.168.2.246:51284 )
  INFO root [2020-10-24 23:00:38,288] - NetworkInterface::initialize(INTERNAL): address 192.168.2.246:51284, SOMAXCONN=128.
  INFO root [2020-10-24 23:00:38,288] - server-configs:
  INFO root [2020-10-24 23:00:38,289] - 	internalAddr : 192.168.2.246:51284
  INFO root [2020-10-24 23:00:38,289] - 	externalAddr : 192.168.2.246:20013
  INFO root [2020-10-24 23:00:38,289] - 	componentID : 920002
  INFO root [2020-10-24 23:00:39,923] - ---- loginapp Version: 1.2.7. ScriptVersion: 0.1.0. Protocol: A87D1428E6DE7B1742DD55BD6E5BAE8C. Config: Debug X86. Built: 19:01:57 Aug  4 2020. AppID: 920002. UID: 1937. PID: 14368 ----
  INFO root [2020-10-24 23:00:39,923] - Server : Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz with 2097151kB RAM
  INFO root [2020-10-24 23:00:39,924] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2020-10-24 23:00:39,924] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
 DEBUG root [2020-10-24 23:00:40,065] - Script::install_py_dlls(): PyInit_pyexpat
 DEBUG root [2020-10-24 23:00:40,066] - Script::install_py_dlls(): PyInit__socket
 DEBUG root [2020-10-24 23:00:40,066] - Script::install_py_dlls(): PyInit__ssl
 DEBUG root [2020-10-24 23:00:40,071] - Script::install_py_dlls(): PyInit__hashlib
 DEBUG root [2020-10-24 23:00:40,071] - Script::install_py_dlls(): PyInit_select
 DEBUG root [2020-10-24 23:00:40,072] - Script::install_py_dlls(): PyInit__ctypes
 DEBUG root [2020-10-24 23:00:40,072] - Script::install_py_dlls(): PyInit__elementtree
 DEBUG root [2020-10-24 23:00:40,117] - Script::install_py_dlls(): PyInit_unicodedata
  INFO root [2020-10-24 23:00:40,154] - Script::install(): is successfully, Python=(3.4.2 (default, Aug  4 2020, 19:01:54) [MSC v.1925 32 bit (Intel)])!
  INFO root [2020-10-24 23:00:40,154] - Script::install(): KBExtra is successfully!
S_INFO root [2020-10-24 23:00:40,156] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2020-10-24 23:00:40,156] - TelnetServer server is running on port 31000
  INFO root [2020-10-24 23:00:40,402] - ---- loginapp is running ----
 DEBUG root [2020-10-24 23:00:40,402] - Components::process(): Request for the process of identity...
  INFO root [2020-10-24 23:00:47,591] - Components::findComponents: find logger(1)...
  INFO root [2020-10-24 23:00:47,692] - Components::findComponents: found logger, addr:192.168.2.246:51278
  INFO root [2020-10-24 23:00:47,692] - Components::addComponent: reset orderLog, uid=1937!
  INFO root [2020-10-24 23:00:47,693] - Components::addComponent[logger], uid=1937, componentID=210002, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2020-10-24 23:00:47,694] - Components::findComponents: find dbmgr(2)...
  INFO root [2020-10-24 23:00:47,700] - Components::findComponents: found logger, addr:192.168.2.246:51278
  INFO root [2020-10-24 23:00:47,704] - Components::findComponents: found dbmgr, addr:192.168.2.246:51279
  INFO root [2020-10-24 23:00:47,704] - Components::addComponent: reset orderLog, uid=1937!
  INFO root [2020-10-24 23:00:47,705] - Components::addComponent[dbmgr], uid=1937, componentID=42002, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2020-10-24 23:00:47,813] - Components::findComponents: found dbmgr, addr:192.168.2.246:51279
  WARN root [2020-10-24 23:00:48,815] - Forwarding logs to logger[192.168.2.246:51278/0/0/0]...
 ERROR root [2020-10-24 23:57:22,960] - Components::removeComponentByChannel: logger : 210002, Abnormal exit! TCPPacketReceiver::processRecv(): errno=10054

  INFO root [2020-10-24 23:57:22,960] - The following logs sent to logger failed:
  WARN root [2020-10-24 23:57:22,960] - ==> [2020-10-24 23:57:22 965] TCPPacketReceiver::processPendingEvents(192.168.2.246:51278): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2020-10-24 23:57:22,960] - ==> [2020-10-24 23:57:22 967] Components::removeComponentByChannel: logger : 210002, Abnormal exit! TCPPacketReceiver::processRecv(): errno=10054

  WARN root [2020-10-24 23:57:23,038] - TCPPacketReceiver::processPendingEvents(192.168.2.246:51279): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2020-10-24 23:57:23,038] - Components::removeComponentByChannel: dbmgr : 42002, Abnormal exit! TCPPacketReceiver::processRecv(): errno=10054

  INFO root [2020-10-24 23:57:23,038] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2020-10-24 23:57:23,038] - onLoginAppShutDown()
  INFO root [2020-10-24 23:57:23,038] - Shutdowner::onShutdown: shutting down
  INFO root [2020-10-24 23:57:23,038] - Shutdowner::onShutdownEnd: shutting down
 DEBUG root [2020-10-24 23:57:23,054] - Script::uninstall_py_dlls(): PyInit_pyexpat
 DEBUG root [2020-10-24 23:57:23,054] - Script::uninstall_py_dlls(): PyInit__socket
 DEBUG root [2020-10-24 23:57:23,054] - Script::uninstall_py_dlls(): PyInit__ssl
 DEBUG root [2020-10-24 23:57:23,054] - Script::uninstall_py_dlls(): PyInit__hashlib
 DEBUG root [2020-10-24 23:57:23,054] - Script::uninstall_py_dlls(): PyInit_select
 DEBUG root [2020-10-24 23:57:23,054] - Script::uninstall_py_dlls(): PyInit__ctypes
 DEBUG root [2020-10-24 23:57:23,054] - Script::uninstall_py_dlls(): PyInit__elementtree
 DEBUG root [2020-10-24 23:57:23,054] - Script::uninstall_py_dlls(): PyInit_unicodedata
  INFO root [2020-10-24 23:57:23,085] - Script::uninstall(): is successfully!
 DEBUG root [2020-10-24 23:57:23,085] - ThreadPool::destroy(): starting size 2.
  WARN root [2020-10-24 23:57:23,398] - ThreadPool::destroy(): waiting for thread(2)[0x1c75970,0x1c75908,], try=1
